### PR TITLE
HDDS-10510. Improve user experience with containerbalancer CLI

### PR DESCRIPTION
**Pull request by sarvekshayr**
## What changes were proposed in this pull request?
`ozone admin containerbalancer stop` command seems stuck and does not return anything while it is in STOPPING state waiting for the current iteration to complete. 
Print progress message to improve the logs for containerbalancer CLI.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-10510

## How was this patch tested?
Manually tested this patch using a docker cluster. Also got a clean CI run.

Screenshot 2024-03-26 at 2 16 42 PM

### CODE REVIEW

**Comment by sarvekshayr** [comment]
@SaketaChalamchala please review this patch.

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStopSubcommand.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 public class ContainerBalancerStopSubcommand extends ScmSubcommand {
   @Override
   public void execute(ScmClient scmClient) throws IOException {
+    System.out.println("ContainerBalancer is in stopping state. Waiting for current iteration to finish");
     scmClient.stopContainerBalancer();
     System.out.println("Stopping ContainerBalancer...");
```
Comment (adoroszlai) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStopSubcommand.java:39]: `...` indicates the command is doing something (waiting for the server).  I think it should appear before sending the stop command.

```
ContainerBalancer is in stopping state. Waiting for current iteration to finish...
ContainerBalancer stopped.
```

or even (printed in two steps):

```
Stopping ContainerBalancer... done.
```

Comment (SaketaChalamchala) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStopSubcommand.java:39]: > `...` indicates the command is doing something (waiting for the server). I think it should appear before sending the stop command.
> 
> ```
> ContainerBalancer is in stopping state. Waiting for current iteration to finish...
> ContainerBalancer stopped.
> ```

+1

**Review by tanvipenumudy — APPROVED** [review]
Thank you @sarvekshayr for the patch. 

+1 pending CI report.

**Review by Tejaskriya — APPROVED** [review]
LGTM +1

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStopSubcommand.java (hunk 1 above)
Comment (siddhantsangwan) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStopSubcommand.java]: NIT: Maybe I'm being overly technical, but how about:
```
Sending stop command. Waiting for Container Balancer to stop...
```
and then 
```
Container Balancer stopped.
```
Balancer doesn't transition to the `STOPPING` state until it has received the stop command, done some validations (this scm is the leader, balancer is currently `RUNNING` etc) and persisted some configurations to RocksDB.

**Review by siddhantsangwan — APPROVED** [review]
@sarvekshayr Thanks for the update. LGTM, pending green CI.

**Comment by adoroszlai** [comment]
Thanks @sarvekshayr for the patch, @SaketaChalamchala, @siddhantsangwan, @tanvipenumudy, @Tejaskriya for the review.

**Pull request merged**